### PR TITLE
[auto parallel] Lazy init with random control.

### PR DESCRIPTION
**Pull request by FeixLiu**
### PR types
Others

### PR changes
Others

### Description
[auto parallel] Lazy init with random control for Replicate.
PCard-73145

Has tested xavier initializer and constant initializer and Replicate status.

### CODE REVIEW

File: test/auto_parallel/test_semi_auto_parallel_single_strategy.py [hunk 1]
```diff
@@ -99,6 +99,16 @@ def test_shard_optimizer_api(self):
...
+    def test_lazy_init(self):
+        envs_list = test_base.gen_product_envs_list(
+            self._default_envs, self._changeable_envs
+        )
+        for envs in envs_list:
+            self.run_test_case(
+                "semi_auto_parallel_lazy_init.py",
+                user_defined_envs=envs,
+            )
+
```
Comment (LiYuRio) [inline test/auto_parallel/test_semi_auto_parallel_single_strategy.py]: 不要在这里加测试了

File: python/paddle/distributed/auto_parallel/api.py [hunk 2]
```diff
@@ -171,19 +172,49 @@ def shard_tensor(
     # `paddle.to_tensor` supports both dynamic and static mode
     if stop_gradient is None:
         stop_gradient = getattr(data, "stop_gradient", True)
-    tensor = paddle.to_tensor(
-        data, dtype=dtype, place=place, stop_gradient=stop_gradient
-    )
+    if hasattr(data, "_is_initialized") and not data._is_initialized():
```
Comment (LiYuRio) [inline python/paddle/distributed/auto_parallel/api.py:175]: tensor会有没有is_initialized这个属性的情况吗，是不是直接判断data是numpy就行了

File: test/auto_parallel/test_semi_auto_parallel_single_strategy.py (hunk 1 above)
Comment (FeixLiu) [reply test/auto_parallel/test_semi_auto_parallel_single_strategy.py]: ok

**Review by LiYuRio — APPROVED** [review]

**Review by JZ-LIANG — APPROVED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]
LGTM

**Pull request merged**